core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -228,31 +223,41 @@ class MapUnit {
         return movement
     }
 
-    fun getUnitToUpgradeTo(): BaseUnit {
+    fun getUnitToUpgradeTo(fromRuins: Boolean): BaseUnit {
         var unit = baseUnit()
+        
+        // Pathfinder can only be upgraded via ruins
+        if (unit.name == "Pathfinder" && !fromRuins) {
```

Comment:
If this is a unique trait of the pathfinder, it should be in the unit's "uniques", and that's what we should be checking